compiler/optimizer/OMRValuePropagation.cpp

```diff
@@ -6476,7 +6476,7 @@ void OMR::ValuePropagation::buildBoundCheckComparisonNodes(BlockVersionInfo *blo
                   // Have we already generated an instanceOf for this object/class when handling array indexes?
                   for (arrayIndex = arrayLength->_arrayIndicesInfo->getFirst(); result != TR_yes && arrayIndex; arrayIndex = arrayIndex->getNext())
                      {
-                     if (arrayIndex->_versionBucket && arrayIndex->_baseNode->getOpCode().getOpCodeValue() == TR::iloadi && 
+                     if (arrayIndex->_versionBucket && arrayIndex->_baseNode && arrayIndex->_baseNode->getOpCode().getOpCodeValue() == TR::iloadi &&
```

Comment:
if arrayIndex->_baseNode is NULL do we have to abandon the versioning? I can see this will stop the crash, but I'm not sure of the semantics... Could you give a bit of context on why this is the right fix rather than also having a case to handle arrayIndex->_baseNode being NULL?

Reply:
If arrayIndex->_baseNode is NULL, does that mean the array index is a direct load or a constant ? Or does it imply some other condition is true wrt the array index ?

Reply:
The purpose of this 'for' loop is to determine if we previously generated an 'instanceOf' for the same class/object combination (and therefore can skip making a redundant one here). The tests to determine if we previously generated an 'instanceOf' needs to be the same as test used previously to decide to generate an instanceOf. I had missed the _baseNode null test and that lead to the build-break as well as possibly allowing for a failure to generate a required instanceOf test for platforms where address 0 is readable.

Reply:
> If arrayIndex->_baseNode is NULL, does that mean the array index is a direct load or a constant ?

In the case where I had recreated the problem it was a constant:

```
n307n     BNDCHK [#1]                                                                         [     0x3ff688cffb0] bci=[2,56,1365] rc=0 vc=312 vn=- li=- udi=- nc=2
n306n       arraylength (stride 2) (X>=0 cannotOverflow )                                     [     0x3ff688cff60] bci=[2,56,1365] rc=2 vc=312 vn=- li=- udi=- nc=1 flg=0x1100
n302n         aloadi  java/lang/String.value [C[#397  final Shadow +8] [flags 0x400a0607 0x0 ] (X!=0 sharedMemory )  [     0x3ff688cfe20] bci=[2,52,1365] rc=3 vc=312 vn=- li=- udi=- nc=1 flg=0x4
n363n           ==>aload
n353n       ==>iconst 0
```

Reply:
I don't think I clearly answered @andrewcraik question. 

If there is no _baseNode then we are not loading the index from an object field and therefore there is no object to perform an instanceOf check against (there is no previously generated instance). So, yes, we should not be skipping the instanceOf since there is no previous instanceOf to make this new one redundant.

Hope that's a more clear answer.

Reply:
ok I think that makes sense - @vijaysun-omr can you approve if you agree? I am going to start the builds.

Reply:
Yes I approve this given Kevin's clarification